internal/handlers/common/error.go

```diff
@@ -249,13 +249,44 @@ func (we *WriteErrors) Document() *types.Document {
 	return d
 }
 
+// parseArr converts err to a T type.
+// If it fails, the returned bool value is false.
+func parseErr[T any](err error) (T, bool) {
```

Comment:
Please use a regular `errors.As` and remove that helper that mixes interfaces with generics